### PR TITLE
Consolidate DisplayLinesWithAlignment into DisplayLines in TextZipper

### PR DESCRIPTION
**Pull request by pdlla**
To avoid breaking changes, a new set of types/methods were created for adding alignment support to TextZipper

Since we are releasing breaking changes in 2.0 this is a good opportunity to keep is as a single set of methods. Note the number of breaking changes that are likely to effect existing usage is probably just `_displayLines_cursorY` being replaced by `_displayLines_cursorPos`

- Consolidate DisplayLinesWithAlignment into DisplayLines
- fix a bug in wrapWithOffsetAndAlignment and add tests
- ran hlint on everything (oops)
- adjust cursor tag position for TextAlignment_Right

### CODE REVIEW

**Comment by pdlla** [comment]
Could someone review. 

I found an unrelated bug in TextZipper that I'm fixing on top of this branch so it would be great to have this merged.

File: src/Data/Text/Zipper.hs [hunk 1]
```diff
@@ -527,7 +378,7 @@ wrapWithOffsetAndAlignment alignment maxWidth n txt = assert (n <= maxWidth) r w
 
 -- converts deleted eol spaces into logical lines
 eolSpacesToLogicalLines :: [[WrappedLine]] -> [[(Text, Int)]]
-eolSpacesToLogicalLines = fmap (fmap (\(WrappedLine a _ c) -> (a,c))) . join . fmap (L.groupBy (\(WrappedLine _ b _) _ -> not b))
+eolSpacesToLogicalLines = fmap (fmap (\(WrappedLine a _ c) -> (a,c))) . ((L.groupBy (\(WrappedLine _ b _) _ -> not b)) =<<)
```
Comment (cgibbard) [inline src/Data/Text/Zipper.hs:381]: Maybe clearer just to use concatMap? This is the list monad, right?

File: src/Data/Text/Zipper.hs [hunk 2]
```diff
@@ -514,7 +365,7 @@ wrapWithOffsetAndAlignment
   -> [WrappedLine] -- (words on that line, hidden space char, offset from beginning of line)
 wrapWithOffsetAndAlignment _ maxWidth _ _ | maxWidth <= 0 = []
 wrapWithOffsetAndAlignment alignment maxWidth n txt = assert (n <= maxWidth) r where
-  r' = splitWordsAtDisplayWidth maxWidth $ T.replicate n " " : wordsWithWhitespace txt
+  r' = splitWordsAtDisplayWidth maxWidth $ wordsWithWhitespace ( T.replicate n "." <> txt)
```
Comment (cgibbard) [inline src/Data/Text/Zipper.hs:368]: What's going on here? Looks like spaces got replaced with dots?

Comment (pdlla) [reply src/Data/Text/Zipper.hs:368]: The leading spaces was causing issues when passed into `splitWordsAtDisplayWidth` so I pad with a non-space character instead. These characters get dropped a few lines later so it doesn't matter what they are. I will add a brief comment explaining this.

File: src/Data/Text/Zipper.hs (hunk 1 above)
Comment (pdlla) [reply src/Data/Text/Zipper.hs:381]: 👍   weird hlint changes (I ran it on the whole file by accident)

**Comment by pdlla** [comment]
Can we get this reviewed merged soon? Is this PR missing anything?
I'd like to build text selection capability into TextZipper after this PR is merged

**Comment by ali-abrar** [comment]
@pdlla Thanks again.

**Pull request merged**